New model of table for UW demo.

sniyaz opened the pull request:
New URDF based off of the old table. Dimensions taken from: https://www.amazon.com/Office-Star-Multipurpose-Rectangle-Folding/dp/B0015TUXT8/ref=sr_1_7?ie=UTF8&qid=1511197564&sr=8-7&keywords=folding+table

Since we don't care too much about the legs, the collision geometry is just a bounding box around the entire table. I've also updated the visual aspects to match the new table dimensions.

**Note:** Since the collision geometry is now a bounding box, the table will have to be moved away from HERB slightly in simulation and the demo (the hands currently need to go under the table in the demo, so it fails with the new table.)

**Test Plan**

Image of new table looking somewhat sane:

![uw_demo_table_herb](https://user-images.githubusercontent.com/8022223/33519965-90afa50c-d77f-11e7-9352-df8e99e750fe.png)

evil-sherdil committed:
Started model of UW demo table.

evil-sherdil committed:
Remove other collision groups since we just use a giant bounding box around the table.

siddhss5 commented:
I don’t like this at all. Why is the collision geometry the entire bounding. I’d instead of composed of a set of primitives for the surface and the legs? It will inevitably result in false positive collisions when we move the base close to the robot, and it’s something no one will fix in the future.

Sidd.

(Rapidly written in motion)

> On Dec 2, 2017, at 1:41 PM, Sherdil Niyaz <notifications@github.com> wrote:
> 
> New URDF based off of the old table. Dimensions taken from: link
> 
> Since we don't care too much about the legs, the collision geometry is just a bounding box around the entire table. I've also updated the visual aspects to match the new table dimensions.
> 
> Note: Since the collision geometry is now a bounding box, the table will have to be moved away from HERB slightly in simulation and the demo (the hands currently need to go under the table in the demo, so it fails with the new table.)
> 
> Test Plan
> 
> Image of new table looking somewhat sane:
> 
> 
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/personalrobotics/pr-ordata/pull/27
> 
> Commit Summary
> 
> Started model of UW demo table.
> Remove other collision groups since we just use a giant bounding box around the table.
> File Changes
> 
> A data/furniture/uw_demo_table.urdf (96)
> Patch Links:
> 
> https://github.com/personalrobotics/pr-ordata/pull/27.patch
> https://github.com/personalrobotics/pr-ordata/pull/27.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

gilwoolee commented:
Sorry Sidd, I somehow thought that was your suggestion at some point and
asked Sherdil to do it.

I think it's at least good to have two geometries, one for collision made
with primitive shapes and the other one with fancy mesh/texture for
rendering.

Sorry for the trouble @Sherdil!

By the way, we used to have "padding" geometry, and
"padding_only"-enable/disable option in prpy, which might be good for being
"just a bit" conservative. I think this requires having additional
"padding" links on the objects. @siddh @mkoval should we port these to
aikido? Were the padding links manually added?



On Sat, Dec 2, 2017 at 3:02 PM, Siddhartha Srinivasa <
notifications@github.com> wrote:

> I don’t like this at all. Why is the collision geometry the entire
> bounding. I’d instead of composed of a set of primitives for the surface
> and the legs? It will inevitably result in false positive collisions when
> we move the base close to the robot, and it’s something no one will fix in
> the future.
>
> Sidd.
>
> (Rapidly written in motion)
>
> > On Dec 2, 2017, at 1:41 PM, Sherdil Niyaz <notifications@github.com>
> wrote:
> >
> > New URDF based off of the old table. Dimensions taken from: link
> >
> > Since we don't care too much about the legs, the collision geometry is
> just a bounding box around the entire table. I've also updated the visual
> aspects to match the new table dimensions.
> >
> > Note: Since the collision geometry is now a bounding box, the table will
> have to be moved away from HERB slightly in simulation and the demo (the
> hands currently need to go under the table in the demo, so it fails with
> the new table.)
> >
> > Test Plan
> >
> > Image of new table looking somewhat sane:
> >
> >
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/personalrobotics/pr-ordata/pull/27
> >
> > Commit Summary
> >
> > Started model of UW demo table.
> > Remove other collision groups since we just use a giant bounding box
> around the table.
> > File Changes
> >
> > A data/furniture/uw_demo_table.urdf (96)
> > Patch Links:
> >
> > https://github.com/personalrobotics/pr-ordata/pull/27.patch
> > https://github.com/personalrobotics/pr-ordata/pull/27.diff
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/personalrobotics/pr-ordata/pull/27#issuecomment-348726677>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7LaA8HoolCDnmAO1BzSOJHFtMB10mTks5s8db4gaJpZM4Qzeir>
> .
>

yskim041 commented:
I agree that we eventually should have a precise collision geometry for the table, but I think that we should keep this version of urdf as well for the table with a tablecloth.

sniyaz commented:
@yskim041, by keep this version do you mean the one with the bounding box as the collision geometry? 

I'm also happy to implement the option mentioned by @gilwoolee, where we still have conservative collision geometry, but still less so than a bounding box.

mkoval commented:
@siddhss5 was suggesting making the simplified collision geometry out of _multiple_ primitive shapes, e.g. one box for the tabletop and a separate box (or cylinder) for each leg. We'd still keep the fancy (high-poly, possibly textured) mesh for visualization.

I think this is consistent with what @gilwoolee suggested above: 🙂  

> I think it's at least good to have two geometries, one for collision made with primitive shapes and the other one with fancy mesh/texture for rendering.

---

Padding is a separate topic. 😉 

> I think this requires having additional "padding" links on the objects. Should we port these to Aikido? Were the padding links manually added?

Yes, this was implemented in PrPy by manually defining additional links. By convention, we prefixed the name of all padding links with the name `padding_` to differentiate them from normal links. I don't think this is necessarily the way we want to handle this in Aikido/DART.

We will likely need this functionality at some point, but there are a lot of details to work out. So I suggest tabling (pardon the pun :trollface:) that discussion for now.

evil-sherdil committed:
Changed table collision geometry to three rectangles to avoid use of a bounding box.

evil-sherdil committed:
Modifiy visual geometry to match collision geometry better.

sniyaz commented:
Changed the collision geometry to no longer be a bounding box, and just a very simple set of three rectangles. I used a tape measure to make sure these dimensions line up with the new table. Also updated the visual geometry slightly to match the new table better.

**Visualization of new collision geometry**

![table_v2_coll_space](https://user-images.githubusercontent.com/8022223/33532130-d4f54ff6-d864-11e7-9d6b-9d9ecd148922.png)

**Visual Geometry**

![uw_table_v2](https://user-images.githubusercontent.com/8022223/33532134-e4cd4c1c-d864-11e7-8710-dd45bd08a909.png)

With these changes, HERB can get his hands under the table in simulation.
